### PR TITLE
 MAP-1458 314 updateContactDetails getReferenceById EntityNotFoundException 


### PR DESCRIPTION
**Pull request by marcusaleman**
before
image

after test1 EntityNotFoundException
image

after test2 ContactDetailNotFoundException
image

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt [hunk 1]
```diff
@@ -342,4 +344,35 @@ class PrisonServiceTest {
...
+    private val prisonRepository2: PrisonRepository = mock()
+    private val contactDetailsRepository2: ContactDetailsRepository = mock()
+
+    @Test
+    fun `should throw ContactDetailsNotFoundException when getByPrisonIdAndType contactDetail not found`() {
+      val removeIfNull = true
+      val updateRequest = ContactDetailsDto(VIDEOLINK_CONFERENCING_CENTRE, emailAddress = "xxx@moj.gov.uk", phoneNumber = "01234567899", webAddress = "www.xxxmojdigital.blog.gov.uk")
+      val prison = Prison("BRI", "Bri Prison", active = true)
+
+      whenever(prisonRepository.getReferenceById(any())).thenReturn(prison)
```
Comment (jnioche-jd) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt:360]: you are using _prisonRepository_ but declared _ prisonRepository2_ is that on purpose?

Comment (jnioche-jd) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt:360]: why do you need to create new variables?

**Review by jnioche-jd — COMMENTED** [review]
couple of questions

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt (hunk 1 above)
Comment (marcusaleman) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt:360]: prisonRepository2 is used in the next Test, line 367.  When you use mocks they can retain the previos state and i got a different behaviour result so I considered create a new one.

Comment (marcusaleman) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonServiceTest.kt:365]: I have added a commit to reset repository mocks beforeEach this fix the issue of reusing them. this apply to all 3 PR opened with similar tests.

**Pull request merged**